app/bundles/EmailBundle/Resources/views/Email/_list.html.twig

```diff
@@ -7,8 +7,24 @@
                     'checkall'        : 'true',
                     'routeBase'       : 'email',
                     'templateButtons' : {
-                        'delete' : permissions['email:emails:deleteown'] or permissions['email:emails:deleteother']
+                        'delete' : permissions['email:emails:deleteown'] or permissions['email:emails:deleteother'],
                     },
+                    'customButtons': [
+                        {
+                            'attr': {
+                                'class': 'btn btn-default btn-sm btn-nospin',
+                                'data-toggle': 'ajaxmodal',
+                                'data-target': '#MauticSharedModal',
+                                'href': path('mautic_email_batch_categories_view'),
+                                'data-header': 'mautic.email.batch.categories'|trans,
+                            },
+                            'btnText': 'mautic.email.batch.categories'|trans,
+                            'iconClass': 'ri-folder-line',
+                        },
+                    ],
+                    'langVar': 'lead.lead',
+                    'routeBase': 'contact',
```

Comment:
@escopecz this seems to be the specific line that makes the Delete button delete contacts instead of emails